src/datumaro/plugins/framework_converter.py

```diff
@@ -0,0 +1,176 @@
+                label = [
+                    ann.label
+                    for ann in dataitem.annotations
+                    if ann.type == TASK_ANN_TYPE[self.task]
+                ][0]
+            elif self.task in ["detection", "instance_segmentation"]:
+                label = [
+                    ann.as_dict()
+                    for ann in dataitem.annotations
+                    if ann.type == TASK_ANN_TYPE[self.task]
+                ]
+            elif self.task == "semantic_segmentation":
+                label = [
+                    ann.image
+                    for ann in dataitem.annotations
+                    if ann.type == TASK_ANN_TYPE[self.task]
+                ][0]
+
+            if self.transform:
+                image = self.transform(image)
+
+            if self.target_transform:
+                label = self.target_transform(label)
+
+            return image, label
+
+except ImportError:
+
+    class DmTorchDataset:
+        def __init__(self):
+            raise ImportError("PyTorch package not found. Cannot convert to PyTorch dataset.")
+
+
+try:
+    import tensorflow as tf
+
+    class DmTfDataset:
+        def __init__(
+            self,
+            dataset: Dataset,
+            subset: str,
+            task: str,
+            output_signature: Optional[tuple] = None,
+        ):
+            self.dataset = dataset.get_subset(subset)
+            self.task = task
+            self.output_signature = output_signature
+
+        def generator_wrapper(self):
+            for item in self.dataset:
+                image = item.media.data
+
+                if self.task == "classification":
+                    label = [
+                        ann.label
+                        for ann in item.annotations
+                        if ann.type == TASK_ANN_TYPE[self.task]
+                    ][0]
+                elif self.task in ["detection", "instance_segmentation"] and isinstance(
+                    self.output_signature[1], dict
+                ):
+                    label = {}
+                    for key, spec in self.output_signature[1].items():
+                        label[key] = tf.convert_to_tensor(
+                            [
+                                ann.as_dict().get(spec.name, None)
+                                for ann in item.annotations
+                                if ann.type == TASK_ANN_TYPE[self.task]
+                            ]
+                        )
+                elif self.task == "semantic_segmentation":
+                    label = [
+                        ann.image
+                        for ann in item.annotations
+                        if ann.type == TASK_ANN_TYPE[self.task]
+                    ][0]
+
+                yield image, label
+
+        def create(self) -> tf.data.Dataset:
+            tf_dataset = tf.data.Dataset.from_generator(
+                self.generator_wrapper, output_signature=self.output_signature
+            )
```

Comment:
I don't know how it works if
```python
self.create().shuffle()
```

I don't know exactly but I guess it might generate all the items and store them in the memory to shuffle.
Please see this https://medium.com/@acordier/tf-data-dataset-generators-with-parallelization-the-easy-way-b5c5f7d2a18.

We might keep the `id`s of items and use `.map()` function to reconstruct the item from the `id`.

Reply:
If we follow this way, we can make a common base class for `torch` and `tf` such as
```python
class Base:
     def _gen_item(self, idx: int):
class Torch:
      def __getitem__(self, idx: int):
            reutrn self._gen_tiem
class TF:
      def create(self):
               return tf.data.Dataset.from_tensors(item_ids).map(lambda idx: self._gen_item(idx))
```

Reply:
fixed at c9a2356f47ecc2117fd76ae8d62b4bc3f01da149. Thanks.